markdown.dtx

```diff
@@ -28024,23 +28044,30 @@ function M.reader.new(writer, options)
       else
         char_length = pos + 1
       end
-      c = lpeg.match({ C(utf8_by_byte_count(char_length)) },s,i+pos)
-      if (c ~= nil) and (unicode.utf8.match(c, chartype)) then
-        return i
+
+      if (chartype == "punctuation") then
+        if lpeg.match(punctuation_by_size[char_length], s, i+pos) then
+          return i
+        end
+      else
+        c = lpeg.match({ C(utf8_by_byte_count(char_length)) },s,i+pos)
+        if (c ~= nil) and (unicode.utf8.match(c, chartype)) then
+          return i
+        end
       end
     end
   end
 
   local function check_preceding_unicode_punctuation(s, i)
-    return check_unicode_type(s, i, -4, -1, "%p")
+    return check_unicode_type(s, i, -4, -1, "punctuation")
```

Comment:
Why do we need to change `%p` (corresponds to [Unicode categories](http://www.unicode.org/reports/tr44/tr44-6.html#Property_Values) P*: Pc+Pd+Ps+Pe+Pi+Pf+Po) to `punctuation`? The signature and contract of the function `check_unicode_type()` seems quite opaque even without overloading the meaning of the parameter `chartype`.

Reply:
As discussed in https://github.com/Witiko/markdown/pull/416/files#r1534218927. Closing here.